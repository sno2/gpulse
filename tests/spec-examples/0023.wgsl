var<private> age: i32;
fn get_age() -> i32 {
  // The type of the expression in the return statement must be 'i32' since it
  // must match the declared return type of the function.
  // The 'age' expression is of type ref<private,i32,read_write>.
  // Apply the Load Rule, since the store type of the reference matches the
  // required type of the expression, and no other type rule applies.
  // The evaluation of 'age' in this context is the i32 value loaded from the
  // memory locations referenced by 'age' at the time the return statement is
  // executed.
  return age;
}

fn caller() {
  age = 21;
  // The copy_age constant will get the i32 value 21.
  let copy_age: i32 = get_age();
}
