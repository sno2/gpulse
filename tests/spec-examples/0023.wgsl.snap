ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 13, .end = 16 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .identifier = ast.Span{ .start = 18, .end = 21 } }, .value = null } }
ast.Node{ .fn_decl = ast.FnDecl{ .name = ast.Span{ .start = 26, .end = 33 }, .params = {  }, .ret = ast.Node{ .identifier = ast.Span{ .start = 39, .end = 42 } }, .scope = { ast.Node{ .ret = ast.Ret{ .value = ast.Node{ .identifier = ast.Span{ .start = 574, .end = 577 } } } } } } }
ast.Node{ .fn_decl = ast.FnDecl{ .name = ast.Span{ .start = 585, .end = 591 }, .params = {  }, .ret = null, .scope = { ast.Node{ .assign = ast.BinaryOp{ .lhs = ast.Node{ .identifier = ast.Span{ .start = 598, .end = 601 } }, .op_idx = 602, .rhs = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 604, .end = 606 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }, ast.Node{ .let_decl = ast.LetDecl{ .name = ast.Span{ .start = 668, .end = 676 }, .typ = ast.Node{ .identifier = ast.Span{ .start = 678, .end = 681 } }, .value = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 684, .end = 691 } }, .args = {  } } } } } } } }
